contacts/api.py

```diff
@@ -0,0 +1,153 @@
+    try:
+        if 'token' in request.COOKIES:
+            token = Token.objects.get(key=request.COOKIES['token'])
+            return token
+        else:
+            return None
+    except Token.DoesNotExist:
+        return None
+
+
+class UserProfile(APIView):
+    permission_classes = (IsUserLogin,)
+
+    def get(self, request):
+        if validate_token(request) is not None:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            serializer = UserSerializer(user)
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("Your token expire Login again", status=HTTP_400_BAD_REQUEST)
+
+    def put(self, request):
+        if validate_token(request) is not None:
+            pk = request.COOKIES.get('user_id')
+            user = User.objects.get(id=pk)
+            payload = request.data
+            serializer = UserSerializer(user, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("UNAUTHORIZED", status=HTTP_401_UNAUTHORIZED)
+
+    def get_serialized(self, pk, user):
+
+        queryset = Contact.objects.filter(user_id=pk)
+        serializer = ContactSerializer(queryset, many=True)
+        return serializer.data
+
+
+class UserContacts(APIView):
+    permission_classes = (IsUserLogin,)
+
+    def get(self, request):
+        if validate_token(request) is not None:
+            pk = request.COOKIES.get('user_id')
+            _id = int(pk)
+            user = User.objects.get(id=_id)
+            contact = Contact.objects.filter(user_id=user.id)
+            serializer = ContactSerializer(contact, many=True)
+            return Response(serializer.data)
+        else:
+            return Response("UNAUTHORIZED", status=HTTP_401_UNAUTHORIZED)
+
+    def post(self, request):
+        serializer = ContactSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors)
+
+    def put(self, request):
+        if validate_token(request) is not None:
+            contact = Contact.objects.get(id=request.data['id'])
+            payload = request.data
+            serializer = ContactSerializer(contact, data=payload, partial=True)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=HTTP_200_OK)
+        else:
+            return Response("UNAUTHORIZED", status=HTTP_401_UNAUTHORIZED)
+
+
+class ContactDetails(APIView):
+
+    def get(self, request, pk):
+        book = Contact.objects.get(id=pk)
+        serializer = ContactSerializer(book)
+        return Response(serializer.data)
+
+
+class UserDetails(APIView):
+
```

Comment:
Use proper PEP8 formatting. Also write docstrings for these methods.

Reply:
Done